src/main/scala/com/databricks/labs/overwatch/pipeline/PipelineFunctions.scala

```diff
@@ -151,6 +151,42 @@ object PipelineFunctions {
       .alias(defaultAlias)
   }
 
+  def optimizeDFForWrite(
+                        df: DataFrame,
+                        target: PipelineTable
+                        ): DataFrame = {
+
+    var mutationDF = df
+
+    mutationDF = if (target.zOrderBy.nonEmpty) {
+      mutationDF.moveColumnsToFront(target.zOrderBy ++ target.statsColumns)
+    } else mutationDF
```

Comment:
where the data skip columns is set? Or we rely on the default value of 32?

Reply:
I leave it at default but ensure the keys are at the front.